operational/azure/vms_without_managed_disks/azure_vms_without_managed_disks.pt

```diff
@@ -0,0 +1,181 @@
+  set_page_marker do
+    uri true
+  end
+end
+
+###############################################################################
+# Datasources
+###############################################################################
+datasource "all_ds_azure_virtual_machines" do
+  request do
+    auth $azure_auth
+    pagination $azure_pagination
+    host "management.azure.com"
+    path join(["/subscriptions/", $param_azure_sub, "/providers/Microsoft.Compute/virtualMachines"])
+    query "api-version","2018-06-01"
+    header "User-Agent", "RS Policies"
+  end
+end
+
+datasource "ds_azure_virtual_machines" do
+  request do
+    auth $azure_auth
+    pagination $azure_pagination
+    host "management.azure.com"
+    path join(["/subscriptions/", $param_azure_sub, "/providers/Microsoft.Compute/virtualMachines"])
+    query "api-version","2018-06-01"
+    header "User-Agent", "RS Policies"
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "value[*]") do
+      field "id", jmes_path(col_item, "id")
+      field "type", jmes_path(col_item, "type")
+      field "location", jmes_path(col_item, "location")
+      field "name", jmes_path(col_item, "name")
+      field "osDisk", jmes_path(col_item, "properties.storageProfile.osDisk")
+      field "tags", jmes_path(col_item, "tags")
+      field "properties", jmes_path(col_item, "properties")
+    end
+  end
+end
+
+datasource "ds_filtered_virtual_machines" do
+  run_script $js_filter_virtual_machines, $ds_azure_virtual_machines, $param_exclusion_tag_key
+end
+
+###############################################################################
+# Scripts
+###############################################################################
+script "js_filter_virtual_machines", type: "javascript" do
+  parameters "ds_azure_virtual_machines", "param_exclusion_tag_key"
+  result "results"
+  code <<-EOS
+    var results = [];
+    _.each(ds_azure_virtual_machines, function(vm){
+      if (vm.osDisk.vhd != null && vm.osDisk.vhd !== undefined){
+        if (_.has(vm.tags, param_exclusion_tag_key)){
+          // skip due to exclusion tag
+        }else{
+          var resourceGroup = vm.id.split('/')[4]
+          var storageAccount = vm.osDisk.vhd.uri.split('/')[2].split('.')[0]
+          results.push({
+            name: vm.name,
+            resourceGroup: resourceGroup,
+            location: vm.location,
+            storageAccount: storageAccount,
+            osDisk: vm.osDisk
+          })
+        }
+      }
+    })
+EOS
+end
+###############################################################################
+# Policy
+###############################################################################
+
+policy "azure_resource_policy" do
+  validate_each $ds_filtered_virtual_machines do
+    summary_template "{{ len data }} Azure VMs without managed disks"
+    detail_template <<-EOS
+# Azure Resources out of compliance
+| Instance Name | Resource Group | Location | Storage Account | osDisk VHD |
```

Comment:
I was asked to review this. My main question/feedback is: if this is enough information for the user to go find the VM in Azure, then it's enough. Seems like we'd want to show the subscription ID here too, somewhere.